arch/lkl/mm/bootmem.c

```diff
@@ -39,7 +48,7 @@ void __init bootmem_init(unsigned long mem_sz)
 	 */
 	max_low_pfn = virt_to_pfn((void *)memory_end);
 	min_low_pfn = virt_to_pfn((void *)memory_start);
-	memblock_add(memory_start, mem_size);
+	memblock_add(__pa(memory_start), mem_size);
```

Comment:
What if we pass `memory_start` instead of `__pa(memory_start)`? Woudn't that make the physical memory start at CONFIG_LKL_MEMORY_START to avoid the changes in virtiolib?

Reply:
I don't this would work as in FLATMEM config (commented on this above) it is assumed that physical addresses start at `ARCH_PFN_OFFSET << PAGE_SIZE` and `memblock_add` expects physical addresses.